src/libcollectionstest/bit/vec.rs

```diff
@@ -630,6 +630,110 @@ fn test_bit_vec_extend() {
                                  0b01001001, 0b10010010, 0b10111101]));
 }
 
+#[test]
+fn test_bit_vec_append() {
+    // Append to BitVec, that holds a multiple of u32::BITS bits
+    let mut a = BitVec::from_bytes(&[0b10100000, 0b00010010, 0b10010010, 0b00110011]);
+    let mut b = BitVec::new();
+    b.push(false);
+    b.push(true);
+    b.push(true);
+
+    a.append(&mut b);
+
+    assert_eq!(a.len(), 35);
+    assert_eq!(b.len(), 0);
+    assert!(b.capacity() >= 3);
+
+    assert!(a.eq_vec(&[true, false, true, false, false, false, false, false,
+                       false, false, false, true, false, false, true, false,
+                       true, false, false, true, false, false, true, false,
+                       false, false, true, true, false, false, true, true,
+                       false, true, true]));
+
+    // Append to arbitrary BitVec
+    let mut a = BitVec::new();
+    a.push(true);
+    a.push(false);
+
+    let mut b = BitVec::from_bytes(&[0b10100000, 0b00010010, 0b10010010, 0b00110011, 0b10010101]);
+
+    a.append(&mut b);
+
+    assert_eq!(a.len(), 42);
+    assert_eq!(b.len(), 0);
+    assert!(b.capacity() >= 40);
+
+    assert!(a.eq_vec(&[true, false, true, false, true, false, false, false,
+                       false, false, false, false, false, true, false, false,
+                       true, false, true, false, false, true, false, false,
+                       true, false, false, false, true, true, false, false,
+                       true, true,  true, false, false, true, false, true,
+                       false, true]));
+}
```

Comment:
*ponders if it's worth checking that append does the right thing when `a` or `b` is empty.*